Mathlib/GroupTheory/Congruence/Defs.lean

```diff
@@ -98,6 +98,44 @@ a given binary relation."]
 def conGen [Mul M] (r : M → M → Prop) : Con M :=
   ⟨⟨ConGen.Rel r, ⟨ConGen.Rel.refl, ConGen.Rel.symm, ConGen.Rel.trans⟩⟩, ConGen.Rel.mul⟩
 
+section RelFacts
+
+namespace ConGen.Rel
+
+variable {w x y z : M} {r : M → M → Prop} [Mul M]
+
+theorem of_rel (h : r x y) : ConGen.Rel r x y := ConGen.Rel.of _ _ h
+
+lemma of_rel_symm (h : r x y) : ConGen.Rel r y x := (of_rel h).symm
+
+lemma one_step (z w : M) (h : r x y) : ConGen.Rel r (z * x * w) (z * y * w) :=
+    ConGen.Rel.mul (ConGen.Rel.mul (ConGen.Rel.refl _) (of_rel h)) (ConGen.Rel.refl _)
+
+lemma one_step_symm (z w : M) (h : r x y) : ConGen.Rel r (z * y * w) (z * x * w) :=
+    ConGen.Rel.mul (ConGen.Rel.mul (ConGen.Rel.refl _) (of_rel h).symm) (ConGen.Rel.refl _)
```

Comment:
These names look strange (to me) and also they're in a funny place in the file: they seem harder than `append_left`, `mul_rel_right` etc (in fact they're easy corollaries of them, probably the proofs could be made shorter using them?)